app/app/src/main/java/org/omecproject/up4/impl/Up4NorthComponent.java

```diff
@@ -180,21 +183,34 @@ private void translateEntryAndDelete(PiTableEntry entry) throws StatusException
     }
 
     /**
-     * Translate the given logical pipeline table entry to a Up4Service entry insertion call.
+     * Translate the given logical pipeline table entry or meter cell config
+     * to a Up4Service entry apply call.
      *
-     * @param entry The logical table entry to be inserted
-     * @throws StatusException if the entry fails translation or cannot be inserted
+     * @param entry The logical table entry or meter cell config to be applied
+     * @throws StatusException if the entry fails translation or cannot be applied
      */
-    private void translateEntryAndInsert(PiTableEntry entry) throws StatusException {
+    private void translateEntryAndApply(PiEntity entry) throws StatusException {
         log.debug("Translating UP4 write request to fabric entry.");
-        if (entry.action().type() != PiTableAction.Type.ACTION) {
-            log.warn("Action profile entry insertion not supported. Ignoring.");
-            throw UNIMPLEMENTED
-                    .withDescription("Action profile entries not supported by UP4.")
-                    .asException();
-        }
         try {
-            up4Service.apply(up4Translator.up4TableEntryToUpfEntity(entry));
+            switch (entry.piEntityType()) {
+                case TABLE_ENTRY:
+                    PiTableEntry tableEntry = (PiTableEntry) entry;
+                    if (tableEntry.action().type() != PiTableAction.Type.ACTION) {
+                        log.warn("Action profile entry insertion not supported. Ignoring.");
```

Comment:
Not ignoring since we are returning an error
```suggestion
                        log.warn("Action profile entry insertion not supported.");
```